packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/__tests__/useUploadFiles.spec.ts

```diff
@@ -190,6 +193,32 @@ describe('useUploadFiles', () => {
     });
   });
 
+  it('should remove upload after processFile promise rejects', async () => {
+    const processFile: StorageManagerProps['processFile'] = ({ file }) =>
+      Promise.reject({ file, key: 'test.png', error: 'forced test error' });
+
+    const { waitForNextUpdate } = renderHook(() =>
+      useUploadFiles({
+        ...props,
+        isResumable: true,
+        processFile,
+        files: [mockQueuedFile],
+      })
+    );
+
+    waitForNextUpdate();
+
+    await waitFor(() => {
+      expect(mockOnUploadError).toHaveBeenCalledWith('forced test error', {
+        key: 'test.png',
+      });
+    });
+
+    await waitFor(() => {
+      expect(mockRemoveUpload).toHaveBeenCalled();
+    });
+  });
```

Comment:
@calebpollman,  With your comments on my example is this Unit Test incorrect?
Should I only be testing a thrown Exception? or is that a *second* test case?